lib/datasource/gitlab-packages/index.ts

```diff
@@ -0,0 +1,48 @@
+import { GitlabHttp } from '../../util/http/gitlab';
+import * as url from '../../util/url';
+import type { GetReleasesConfig, ReleaseResult } from '../types';
+import type { GitlabPackage } from './types';
+
+const gitlabApi = new GitlabHttp();
+
+export const id = 'gitlab-packages';
+export const customRegistrySupport = true;
+export const registryStrategy = 'first';
+export const defaultVersioning = 'loose';
```

Comment:
Why we use `loose` versioning by default here? No other datasource is setting a default versioning like this 🤔

Reply:
From Gitlab Documentation:

> package_version (String) : The package version. The following regex validates this: `\A(\.?[\w\+-]+\.?)+\z`. You can test your version strings on [Rubular](https://rubular.com/r/aNCV0wG5K14uq8).
https://docs.gitlab.com/ee/user/packages/generic_packages/index.html#publish-a-package-file

![image](https://user-images.githubusercontent.com/6305913/128150795-2a65eb7a-f5d6-44b7-a211-71b6e33cb189.png)

So theses are all valid version for a Gitlab Packages

```
word
single-word
single+word
single.1+word
v1.2.3
v1.2.3-rc4
v.1.2.3+rc5
.55.
.+.
```
So by looking at Renovate versionning documentation, it looked like the `loose` versionning was better than `semver` in this case. But maybe `docker` is better?

Reply:
We have quite a few datasources where there can be non-versions in the values. GitHub tags are a simple example. But we generally take the logic of:
- Don't attempt to upgrade unless the existing value in the repo is a valid version
- Filter out all non-versions from the list of candidates

If you push ahead with `loose` I think the results will be undesirable for most. For example, if the current version is "next" then Renovate would propose to update to "stable", because it sorts later. Or even from "next" to "v1.2.3".

Better to start with docker versioning if it's likely to use similar tagging.